packages/encrypted_storage/lib/src/encrypted_storage.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:encrypted_storage/src/abstract_storage.dart';
+import 'package:encrypted_storage/src/cipher_storage.dart';
+import 'package:encrypted_storage/src/encrypt_helper.dart';
+import 'package:encrypted_storage/src/storage.dart';
+
+/// {@template encrypted_storage}
+/// Encrypted storage
+/// {@endtemplate}
+class EncryptedStorage implements AbstractStorage {
+  /// {@macro encrypted_storage}
+  EncryptedStorage()
+      : _storage = Storage(),
+        _cipherStorage = CipherStorage();
+
+  final CipherStorage _cipherStorage;
+  late EncryptHelper _encryptHelper;
+  final Storage _storage;
+
+  /// Init encrypted storage
+  Future<void> init([String dbName = 'encrypted_storage.db']) async {
+    await Future.wait([
+      _cipherStorage.init(),
+      _storage.init(dbName),
+    ]);
+    _encryptHelper = EncryptHelper(_cipherStorage);
+  }
+
+  /// Reset storage
+  Future<void> reset([String dbName = 'encrypted_storage.db']) async {
+    return _storage.reset(dbName);
+  }
+
+  @override
+  Future<void> clearAll() async {
+    return _storage.clearAll();
+  }
+
+  @override
+  Future<void> clearDomain([String? domain = defaultDomain]) async {
+    return _storage.clearDomain(domain);
+  }
+
+  @override
+  Future<void> delete(String key, {String domain = defaultDomain}) async {
+    return _storage.delete(_encryptHelper.encrypt(key), domain: domain);
+  }
+
+  @override
+  Future<void> deleteDomain(
+    List<String> keys, {
```

Comment:
How can we know list of all keys if it's, for example, nekoton. It will tell only clear store and won't tell you about all keys

Reply:
getDomain(domain) will return all key-value pairs from db. Do you think we should create something like getDomainKeys which will return only keys?

Reply:
I'm not sure this is not a good solution. Maybe we could delete domain only by `domain` if it's possible

Reply:
We already have ability to do this by using clearDomain() :)

Reply:
I just aded getDomainKeys.